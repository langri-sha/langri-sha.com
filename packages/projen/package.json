{
  "name": "@langri-sha/projen",
  "version": "0.1.0",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@langri-sha/tsconfig": "workspace:*",
    "projen": "0.81.1"
  },
  "peerDependencies": {
    "projen": "^0.81.1",
    "typescript": "^5.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
