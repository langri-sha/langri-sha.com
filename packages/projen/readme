# @langri-sha/projen

Collection of [projen] templates for bootstraping monorepos and workspace projects.

[projen]: https://projen.dev/
