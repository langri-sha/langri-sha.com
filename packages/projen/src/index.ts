import {
  Project as BaseProject,
  type ProjectOptions as BaseProjectOptions,
} from 'projen'

export interface ProjectOptions extends BaseProjectOptions {}

export class Project extends BaseProject {
  constructor(options: ProjectOptions) {
    super({
      ...options,
      gitIgnoreOptions: getGitIgnoreOptions(options),
    })
  }
}

const getGitIgnoreOptions = (
  options: ProjectOptions,
): ProjectOptions['gitIgnoreOptions'] => ({
  ...options.gitIgnoreOptions,
  ignorePatterns: [
    ...`
    .*
    !.babelrc
    !.dockerignore
    !.editorconfig
    !.eslintignore
    !.eslintrc.js
    !.gitattributes
    !.gitignore
    !.gitkeep
    !.husky
    !.npmignore
    !.openssl
    !.prettierignore
    !.terraform.lock.hcl
    *.log
    *.tsbuildinfo

    !.github/
    !.husky/
    !.projen/
    dist/
    local/
    node_modules/
    `
      .split('\n')
      .map((pattern) => pattern.trim())
      .filter((pattern) => pattern.length > 0),
    ...(options.gitIgnoreOptions?.ignorePatterns ?? []),
  ],
})
