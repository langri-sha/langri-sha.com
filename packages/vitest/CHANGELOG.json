{
  "name": "@langri-sha/vitest",
  "entries": [
    {
      "date": "Sat, 27 Jul 2024 15:50:17 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d542415a6b2c2c339a8367c77b925842000f9f9a",
            "comment": "feat(workspace): Exclude compiling test modules"
          }
        ]
      }
    },
    {
      "date": "Fri, 26 Jul 2024 21:00:29 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d58e2b444358fb65166d0878c185220c6e0a7bb0",
            "comment": "feat(workspace): Migrate documentation pathnames"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d58e2b444358fb65166d0878c185220c6e0a7bb0",
            "comment": "feat(projen): Add Git ignore patterns"
          }
        ]
      }
    },
    {
      "date": "Wed, 17 Jul 2024 10:48:15 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "patch": [
          {
            "author": "email not defined",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "chore(deps): chore(deps): update dependency vitest to v2"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "fix(workspace): Use correct GitHub directory"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "feat(root): Migrate to ESM"
          }
        ]
      }
    },
    {
      "date": "Mon, 08 Jul 2024 21:43:05 GMT",
      "version": "0.1.0",
      "tag": "@langri-sha/vitest_v0.1.0",
      "comments": {
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Use `vitest`"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(vitest): Add `nock`"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Add repository path"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(tsconfig): Use `${configDir}`"
          }
        ],
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "fix(projen): Use correct output directory"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "fix(projen): Use workspace `@langri-sha/tsconfig`"
          }
        ]
      }
    }
  ]
}
