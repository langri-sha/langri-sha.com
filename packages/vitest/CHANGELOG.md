# Change Log - @langri-sha/vitest

This log was last generated on Wed, 17 Jul 2024 10:48:15 GMT and should not be manually modified.

<!-- Start content -->

## 0.1.1

Wed, 17 Jul 2024 10:48:15 GMT

### Patches

- chore(deps): chore(deps): update dependency vitest to v2 (email not defined)
- fix(workspace): Use correct GitHub directory (filip.dupanovic@gmail.com)

## 0.1.0

Mon, 08 Jul 2024 21:43:05 GMT

### Minor changes

- feat(workspace): Use `vitest` (filip.dupanovic@gmail.com)
- feat(vitest): Add `nock` (filip.dupanovic@gmail.com)

### Patches

- fix(projen): Use correct output directory (filip.dupanovic@gmail.com)
- fix(projen): Use workspace `@langri-sha/tsconfig` (filip.dupanovic@gmail.com)
