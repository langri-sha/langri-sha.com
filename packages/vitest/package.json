{
  "name": "@langri-sha/vitest",
  "version": "0.0.0",
  "bugs": {
    "url": "https://github.com/langri-sha/langri-sha.com/issues"
  },
  "repository": {
    "url": "langri-sha/langri-sha.com",
    "path": "/packages/vitest"
  },
  "license": "MIT",
  "author": {
    "name": "Filip DupanoviÄ‡",
    "email": "filip.dupanovic@gmail.com",
    "url": "https://langri-sha.com",
    "organization": false
  },
  "type": "module",
  "main": "src/index.ts",
  "scripts": {
    "prepublishOnly": "rm -rf dist; tsc --project tsconfig.build.json"
  },
  "dependencies": {
    "nock": "13.5.4",
    "tempy": "1.0.1"
  },
  "devDependencies": {
    "@langri-sha/tsconfig": "workspace:*"
  },
  "peerDependencies": {
    "vitest": "^1.6.0"
  },
  "publishConfig": {
    "access": "public",
    "main": "dist/index.js",
    "types": "dist/index.d.ts"
  },
  "//": "~~ Generated by projen. To modify, edit .projenrc.mts and run \"npx projen\"."
}
