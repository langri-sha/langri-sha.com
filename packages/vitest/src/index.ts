export * from 'vitest'
export { default as nock } from 'nock'
export { default as tempy } from 'tempy'
