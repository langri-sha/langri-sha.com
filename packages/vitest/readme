# @langri-sha/vitest

Provides some useful helpers that are commonly used for authoring tests.

## Usage

Install the required dependencies:

```sh
npm install -D vitest @langri-sha/vitest
```

Then import your Vitest dependencies from here:

```js
// some.test.js
import { expect, test, tempy } from '@langri-sha/vitest'

test(/*...*/)
```

## See

- [`@jest/globals`]
- [`tempy`]

[`@jest/globals`]: https://jestjs.io/docs/api
[`tempy`]: https://github.com/sindresorhus/tempy
