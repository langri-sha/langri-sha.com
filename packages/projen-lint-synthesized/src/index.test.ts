import { directorySnapshot } from 'projen/lib/util/synth'
import { expect, tempy, test } from '@langri-sha/vitest'

import { promises as fs } from 'node:fs'
import path from 'node:path'

import { Project, TextFile } from 'projen'
import { LintSynthesized, type LintSynthesizedOptions } from './index'

const setup = (options?: LintSynthesizedOptions) => {
  const outdir = tempy.directory()

  const project = new Project({
    name: 'test-project',
    outdir,
  })

  project.removeTask('build')
  project.removeTask('compile')
  project.removeTask('eject')
  project.removeTask('default')
  project.removeTask('package')
  project.removeTask('post-compile')
  project.removeTask('pre-compile')
  project.removeTask('test')

  new LintSynthesized(project, options)

  return { project }
}

test('defaults', () => {
  const { project } = setup()

  project.synth()
  expect(directorySnapshot(project.outdir)).toMatchSnapshot()
})

test('lints synthesized files', async () => {
  const { project } = setup({
    '*': 'prettier --ignore-unknown --write',
  })

  const project2 = new Project({
    name: 'test-project2',
    outdir: 'project2',
    parent: project,
  })

  let file, contents

  file = new TextFile(project, 'test1.js')
  file.addLine(`module.exports = ${JSON.stringify({ foo: 'bar' })}`)

  file = new TextFile(project, 'test2.js')
  file.addLine(`module.exports = ${JSON.stringify({ foo: 'bar' })}`)

  file = new TextFile(project2, 'test3.js')
  file.addLine(`module.exports = ${JSON.stringify({ foo: 'bar' })}`)

  project.synth()

  file = await fs.readFile(path.join(project.outdir, 'test1.js'))
  contents = file.toString('utf8')

  expect(contents).toEqual(`module.exports = { foo: "bar" };\n`)

  file = await fs.readFile(path.join(project.outdir, 'test2.js'))
  contents = file.toString('utf8')

  expect(contents).toEqual(`module.exports = { foo: "bar" };\n`)

  file = await fs.readFile(path.join(project.outdir, 'project2', 'test3.js'))
  contents = file.toString('utf8')
})

test('lints hidden synthesized files', async () => {
  const { project } = setup({
    '*': 'prettier --ignore-unknown --write',
  })

  let file

  file = new TextFile(project, '.test.js')
  file.addLine(`module.exports = ${JSON.stringify({ foo: 'bar' })}`)

  project.synth()

  file = await fs.readFile(path.join(project.outdir, '.test.js'))
  const contents = file.toString('utf8')

  expect(contents).toEqual(`module.exports = { foo: "bar" };\n`)
})

test('preserves file modes', async () => {
  const { project } = setup({
    '*': 'prettier --ignore-unknown',
  })

  new TextFile(project, 'test.sh', {
    executable: true,
    readonly: true,
  })

  project.synth()

  await expect(
    fs.stat(path.join(project.outdir, 'test.sh')),
  ).resolves.toHaveProperty('mode', 33_124)
})
