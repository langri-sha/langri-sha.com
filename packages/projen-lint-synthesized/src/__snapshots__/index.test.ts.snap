// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`defaults 1`] = `
{
  ".gitignore": "# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
node_modules/
!/.gitattributes
!/.projen/tasks.json
!/.projen/deps.json
!/.projen/files.json
",
  ".projen/files.json": {
    "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
    "files": [
      ".gitattributes",
      ".gitignore",
      ".projen/deps.json",
      ".projen/files.json",
      ".projen/tasks.json",
    ],
  },
}
`;
